expression/builtin_json.go

```diff
@@ -719,8 +722,42 @@ type jsonMergePreserveFunctionClass struct {
 	baseFunctionClass
 }
 
+type builtinJSONMergePreserveSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJSONMergePreserveSig) Clone() builtinFunc {
+	newSig := &builtinJSONMergePreserveSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
 func (c *jsonMergePreserveFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_MERGE_PRESERVE")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	argTps := make([]types.EvalType, 0, len(args))
+	for range args {
+		argTps = append(argTps, types.ETJson)
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETJson, argTps...)
```

Comment:
I think we don't need to support another function signature, since they share the same code logic. How about return `builtinJSONMergeSig{}` in this function?

Reply:
Thanks, PTAL. And I think the implementation of signature should be placed in `builtinJSONMergePreserveSig`, since `JSON_Merge` is deprecated. Do you think it necessary?